### PR TITLE
GODRIVER-2698 Make Comment fields any-type on all options structs and setters

### PR DESCRIPTION
**Pull request by prestonvasquez**
GODRIVER-2698

## Summary

Update all comment fields on option structs to accept an "any" type that is later parsed to be either a document or a string.

## Background & Motivation

The Comment field on most options structs is an interface. However, on a few structs, Comment is a string because that type was required in an earlier form of the specification. Update all Comment fields and corresponding SetComment setters to be type any.

### CODE REVIEW

**Review by blink1073 — DISMISSED** [review]
LGTM!

File: mongo/collection.go [hunk 1]
```diff
@@ -1671,11 +1677,12 @@ func (coll *Collection) FindOne(ctx context.Context, filter interface{},
 			ShowRecordID:        opt.ShowRecordID,
 			Skip:                opt.Skip,
 			Sort:                opt.Sort,
-		})
+		}
 	}
+
 	// Unconditionally send a limit to make sure only one document is returned and the cursor is not kept open
 	// by the server.
-	findOpts = append(findOpts, options.Find().SetLimit(-1))
```
Comment (matthewdale) [inline mongo/collection.go:1678]: Optional: The explicit indexing is somewhat confusing at first and seems brittle. Is there a reason to do that vs using `append`?

**Review by matthewdale — APPROVED** [review]
Looks good 👍

File: mongo/collection.go (hunk 1 above)
Comment (prestonvasquez) [reply mongo/collection.go:1678]: I can't think of a valid reason not to do it with the append solution. I will update the code, thanks.

Comment (prestonvasquez) [reply mongo/collection.go:1678]: The append method causes the following linting error:

```
mongo/collection.go:1667:14: append to slice `findOpts` with non-zero initialized length (makezero)
                findOpts = append(findOpts, &options.FindOptions{
                           ^
mongo/collection.go:1685:13: append to slice `findOpts` with non-zero initialized length (makezero)
        findOpts = append(findOpts, options.Find().SetLimit(-1))
```

Comment (prestonvasquez) [reply mongo/collection.go:1678]: @matthewdale Looking at this more closely, I believe that the original solution and this solution both have a pretty insidious bug. Namely if a user passes options like this to the FindOne function: [op1, nil, op2], you would get an incorrect merge. In the solution I am proposing you'd actually get a runtime fatal. I've updated the PR with a test for this.

File: mongo/options/aggregateoptions.go [hunk 2]
```diff
@@ -46,7 +46,7 @@ type AggregateOptions struct {
 
 	// A string that will be included in server logs, profiling logs, and currentOp queries to help trace the operation.
 	// The default is nil, which means that no comment will be included in the logs.
```
Comment (matthewdale) [inline mongo/options/aggregateoptions.go:48]: We should update the comments to describe the expected value types. For example, `Comment` fields that are already `interface{}` have the following documentation:

> A string or document that will be included in server logs, profiling logs, and currentOp queries to help trace the operation.  The default is nil, which means that no comment will be included in the logs.

That suggestion applies to all doc comments for the updated `Comment` fields.

**Review by matthewdale — DISMISSED** [review]
Suggested documentation correction. Otherwise, looks good! 👍

**Comment by prestonvasquez** [comment]
@blink1073 @matthewdale Could you re-review this PR?

**Review by blink1073 — DISMISSED** [review]

**Review by matthewdale — APPROVED** [review]
Looks good! 👍

File: mongo/collection.go (hunk 1 above)
Comment (matthewdale) [reply mongo/collection.go:1678]: Nice find!

**Pull request merged**